catalyst-gateway/bin/src/service/api/documents/post_document_index_query/response.rs

```diff
@@ -166,3 +169,36 @@ impl Example for IndexedDocumentVersionDocumented {
         Self(IndexedDocumentVersion::example())
     }
 }
+
+impl TryFrom<SignedDocBody> for IndexedDocumentVersionDocumented {
+    type Error = anyhow::Error;
+
+    fn try_from(doc: SignedDocBody) -> Result<Self, Self::Error> {
+        let mut doc_ref = None;
+        let mut reply = None;
+        let mut template = None;
+        let mut brand = None;
+        let mut campaign = None;
+        let mut category = None;
+        if let Some(json_meta) = doc.metadata() {
+            let meta: catalyst_signed_doc::ExtraFields = serde_json::from_value(json_meta.clone())?;
+            doc_ref = meta.doc_ref().map(Into::into);
+            reply = meta.reply().map(Into::into);
+            template = meta.template().map(Into::into);
+            brand = meta.brand_id().map(Into::into);
+            campaign = meta.campaign_id().map(Into::into);
+            category = meta.campaign_id().map(Into::into);
+        }
+
+        Ok(IndexedDocumentVersionDocumented(IndexedDocumentVersion {
+            ver: DocumentVer::new_unchecked(doc.ver().to_string()),
+            doc_type: DocumentType::new_unchecked(doc.doc_type().to_string()),
```

Comment:
Perhaps I'm missing something, but we can implement `From<Uuid>` for these types and get rid of the `new_unchecked` constructors.

Reply:
We have already `TryFrom<uuid::Uuid>` for these `DocumentVer` and `DocumentType` which makes a conversion with validation and that what I want to avoid at this place. I just want to instantiate  `DocumentVer` and `DocumentType` objects without any underlying validation, by assuming that we have a correct data from the db.